lib/rspec/sidekiq/matchers/have_enqueued_job.rb

```diff
@@ -2,12 +2,13 @@ module RSpec
   module Sidekiq
     module Matchers
       def have_enqueued_job(*expected_arguments)
+        warn "[DEPRECATION] `have_enqueued_job` is deprecated.  Please use `have_enqueued_sidekiq_job` instead."
         HaveEnqueuedJob.new expected_arguments
```

Comment:
We should have this just delegate to `have_enqueued_sidekiq_job` just to make sure they stay consistent. Something like
```ruby
def have_enqueued_job(*args)
  warn 'deprectated'
  have_enqueued_sidekiq_job(args)
end
```